src/polylineencoder.h

```diff
@@ -38,7 +38,32 @@ class PolylineEncoder
 {
 public:
     using Point = std::tuple<double, double>;
-    using Polyline = std::vector<Point>;
+
+    class Polyline
```

Comment:
I understand the intention here: you want to control how new points are created and make sure the coordinates are always rounded. Well, this may work, but the solution itself is overkill. The problem actually lays in the `Point` type, and not in `Polyline` - the container of points. Instead of wrapping `std::vector` in this class with the bunch of functions, you might concentrate on `Point` type. The correct solution would be as follows:
````C++
class Point
{
public:
	Point(double latitude, double longitude)
		:
			m_latitude(std::round(latitude * s_presision) / s_presision),
			m_longitude(std::round(longitude * s_presision) / s_presision)
	{}
	
	double latitude() const { return m_latitude;}
	double longitude() const { return m_longitude;}
private:
	double m_latitude;
	double m_longitude;
};
````
The rest of API (most of) will remain intact. This approach will ensure that the algorithm always processes the rounded values (up to five position after decimal point).

Reply:
I'm absolutely agree with you, the only thing, why i didn't touch the Point is because it thought you wouldn't like that :)

Reply:
I always like reasonable changes:)